src/Diagnostics.DataProviders/KustoHeartBeatService/KustoHeartBeatService.cs

```diff
@@ -0,0 +1,176 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Threading;
+using Diagnostics.Logger;
+using Microsoft.AspNetCore.Http.Features;
+
+namespace Diagnostics.DataProviders
+{
+    public interface IKustoHeartBeatService : IDisposable
+    {
+        string GetClusterNameFromStamp(string stampName);
+    }
+
+    public class KustoHeartBeatService : IKustoHeartBeatService
+    {
+        private KustoDataProviderConfiguration _configuration;
+        Dictionary<string, KustoHeartBeat> _heartbeats;
+        KustoDataProvider _kustoDataProvider;
+
+        List<Thread> _threads;
+
+        private void Initialize()
+        {
+            _heartbeats = new Dictionary<string, KustoHeartBeat>();
+            _kustoDataProvider = new KustoDataProvider(new OperationDataCache(), _configuration, Guid.NewGuid().ToString(), this);
+            _threads = new List<Thread>();
+            foreach (var value in _configuration.FailoverClusterNameCollection)
+            {
+                _heartbeats[value.Key] = new KustoHeartBeat(value.Key, value.Value, _kustoDataProvider);
+                // Start threads for each heartbeat
+                ThreadStart threadStart = new ThreadStart(_heartbeats[value.Key].RunHeartBeatThread);
+                Thread thread = new Thread(threadStart);
```

Comment:
To acheive the same, Should we use Tasks instead of creating new Threads.  That will use TaskScheduler (running on thread using ThreadPool) and every task will not create its own thread